From 7747b3e1ae96273443634bba5c686bd37eb349f0 Mon Sep 17 00:00:00 2001
From: Dmitry Muhomor <muhomor.dmitry@gmail.com>
Date: Tue, 30 Apr 2024 13:26:04 +0300
Subject: [PATCH] disable most of PackageManagerService code for handling
 instant apps

HIDE_EPHEMERAL_APIS flag reduces PackageManagerService attack surface by disabling most of the code
for handling instant apps.
---
 .../core/java/com/android/server/pm/PackageManagerService.java  | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/services/core/java/com/android/server/pm/PackageManagerService.java b/services/core/java/com/android/server/pm/PackageManagerService.java
index 7cffe3829eeb4..def35e58e67db 100644
--- a/services/core/java/com/android/server/pm/PackageManagerService.java
+++ b/services/core/java/com/android/server/pm/PackageManagerService.java
@@ -361,7 +361,7 @@ public class PackageManagerService implements PackageSender, TestUtilityService
 
     static final String SHELL_PACKAGE_NAME = "com.android.shell";
 
-    static final boolean HIDE_EPHEMERAL_APIS = false;
+    static final boolean HIDE_EPHEMERAL_APIS = true;
 
     static final String PRECOMPILE_LAYOUTS = "pm.precompile_layouts";
 
